src/retrie/trie.py

```diff
@@ -35,20 +52,84 @@ class Trie:
 
     __slots__ = "data"
 
-    def __init__(self):
+    def __init__(
+        self, *word  # type: Text
+    ):
         """Initialize data dictionary."""
         self.data = {}  # type: data_type
+        self.add(*word)
+
+    def __eq__(
+        self, other  # type: Any
+    ):  # type: (...) -> bool
+        """Compare two Trie objects."""
+        if not isinstance(other, Trie):
+            return False
+        return self.data == other.data
+
+    def __add__(
+        self, other  # type: "Trie"
+    ):  # type: (...) -> "Trie"
+        """Merge two Trie objects."""
+        if not isinstance(other, Trie):
+            raise TypeError(
+                "Unsupported operand type(s) for +: '{0}' and '{1}'".format(
+                    type(self), type(other)
+                )
+            )
+
+        trie = Trie()
+        trie.data = self._merge(self.data, other.data)
+        return trie
+
+    def __iadd__(
+        self,
+        other,  # type: "Trie"
+    ):  # type: (...) -> "Trie"
+        """Merge another Trie object into the current Trie."""
+        if not isinstance(other, Trie):
```

Comment:
```suggestion
        if self.__class__ != other.__class__:
```
to avoid unintended behaviour with subclasses: `Trie() + SpecialTrie()` should fail